lib/addons/coredns/index.ts

```diff
@@ -1,15 +1,25 @@
-import { CoreAddOn } from "../core-addon";
+import { CoreAddOn, CoreAddOnProps } from "../core-addon";
+
+/**
+ * Configuration options for the coredns add-on.
+ */
+type corednsCollectorAddOnProps = CoreAddOnProps;
```

Comment:
We should not expose the whole CoreAddOnProps for this addon. This will mean that customers will be able to change the name of the addon from coredns to something else, or change the name of SA, which won't work in most cases. So let's exclude name and saName from this, e.g. `Omit<CoreAddOnProps, fields...>`